From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "taehyun.cho" <taehyun.cho@samsung.com>
Date: Thu, 15 Oct 2020 23:16:44 -0700
Subject: FROMLIST: USB: gadget: f_fs: add SuperSpeed Plus support

Setup the descriptors for SuperSpeed Plus for f_fs.

[CPNOTE: 21/05/21] Lee: Functionality found upstream.  Submitted a revert:
      https://android-review.googlesource.com/c/kernel/common/+/1715844

Bug: 170925797
Signed-off-by: taehyun.cho <taehyun.cho@samsung.com>
Link: https://lore.kernel.org/r/20201126180937.255892-3-gregkh@linuxfoundation.org
Change-Id: I261bc3fb6f586b81d2233f60a6fe0a5f250b437b
(cherry picked from commit 00572be28ec474d7953f1b9dd681cc2dd290d9bf)
[willmcvicker: only cherry-picked f_fs portion of original patch]
Signed-off-by: Will McVicker <willmcvicker@google.com>
(cherry picked from commit fc5d4d05b09dcc3d74c77ee6dd3fe5d451364a3b)
Signed-off-by: Will McVicker <willmcvicker@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
---
 drivers/usb/gadget/function/f_fs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/function/f_fs.c b/drivers/usb/gadget/function/f_fs.c
--- a/drivers/usb/gadget/function/f_fs.c
+++ b/drivers/usb/gadget/function/f_fs.c
@@ -3228,6 +3228,10 @@ static int _ffs_func_bind(struct usb_configuration *c,
 	func->function.os_desc_n =
 		c->cdev->use_os_string ? ffs->interfaces_count : 0;
 
+	if (likely(super)) {
+		func->function.ssp_descriptors =
+			usb_copy_descriptors(func->function.ss_descriptors);
+	}
 	/* And we're done */
 	ffs_event_add(ffs, FUNCTIONFS_BIND);
 	return 0;
